src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -488,7 +488,76 @@ public int GetLength(MemoryPoolIterator2 end)
             }
         }
 
-        public string GetString(MemoryPoolIterator2 end)
+        private static unsafe string SingleBlockAsciiString(byte[] input, int offset, int length)
+        {
+            // avoid declaring other local vars, or doing work with stackalloc
+            // to prevent the .locals init cil flag , see: https://github.com/dotnet/coreclr/issues/1279
+            char* output = stackalloc char[length];
```

Comment:
Similar issues with url length: http://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers